File: 프로그래머스/3/43162. 네트워크/네트워크.js
var count = 0;
function dfs(start,n,computers,visited){
    for(let i=0; i<n; i++){
        if(i!=start && !visited[start][i]&&computers[start][i]==1){
            visited[start][i]=true;
            visited[start][start]=true;
            visited[i][start]=true;
            visited[i][i]=true;
            dfs(i,n,computers,visited);
        }
    }
    return;
}

function solution(n, computers) {
    var answer = 0;
    var visited = Array.from(Array(n), () => new Array(n).fill(false));
    for(let i=0; i<n; i++){
        if(!visited[i][i]){
            dfs(i,n,computers,visited);
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    static boolean[] visited;
    public int solution(int n, int[][] computers) {
        visited = new boolean[n];
        int answer = 0;
        for(int i=0; i<n; i++){
            if(!visited[i]) {
                dfs(i,computers);
                answer++;
            }
        }
        return answer;
    }
    static void dfs(int n,int[][] computers){
        visited[n]=true;
        for(int i=0; i<computers.length; i++){
            if(n!=i&&computers[n][i]==1&&!visited[i]){
                dfs(i,computers);
            }
        }
    } 
}